client/src/components/AddProduct/AddProductPreview/AddProductPreview.tsx

```diff
@@ -0,0 +1,103 @@
+import {
+  Grid,
+  CssBaseline,
+  Dialog,
+  DialogContent,
+  Icon,
+  Typography,
+  Button,
+  DialogActions,
+  ListItem,
+  ListItemText,
+  List,
+  CircularProgress,
+} from '@material-ui/core';
+import React, { useState } from 'react';
+import { ProductDetails } from '../../../interface/ProductDetails';
+import useStyles from './useStyles';
+
+interface Props {
+  showPreviewModal: boolean;
+  setShowPreviewModal: React.Dispatch<React.SetStateAction<boolean>>;
+  productDetails: ProductDetails | undefined;
+  handleSubmit: any;
+}
+
+const AddProductPreview = ({
+  showPreviewModal,
+  setShowPreviewModal,
+  productDetails,
+  handleSubmit,
+}: Props): JSX.Element => {
+  const classes = useStyles();
+  const handleCloseProductPreviewModal = () => {
+    setShowPreviewModal(false);
+  };
+
+  const [isSubmitting, setIsSubmitting] = useState(false);
+
+  const submitNewItem = async () => {
+    // send new item to the api
+    setIsSubmitting(true);
+    await handleSubmit();
+    setIsSubmitting(false);
+  };
+
+  if (productDetails != undefined) {
```

Comment:
if productDetails is not defined then i rather render this logic inthe parent component, not in ths component itself and output an empty div element

Reply:
i need help on this